backend/src/main/scala/cromwell/backend/validation/ContinueOnReturnCodeValidation.scala

```diff
@@ -0,0 +1,46 @@
+package cromwell.backend.validation
+
+import cromwell.backend.validation.RuntimeAttributesValidation._
+import cromwell.core._
+import wdl4s.types.{WdlArrayType, WdlStringType}
+import wdl4s.values.{WdlArray, WdlBoolean, WdlInteger, WdlString}
+
+import scalaz.Scalaz._
+
+trait ContinueOnReturnCodeValidation extends RuntimeAttributesValidation[ContinueOnReturnCode] {
+  override def key = RuntimeAttributesKeys.ContinueOnReturnCodeKey
+
+  override def staticDefaultOption = Option(WdlInteger(0))
+
+  override def coercion = ContinueOnReturnCode.validWdlTypes
+
+  override protected def failureMessage = {
+    s"Expecting $key runtime attribute to be either a Boolean, a String 'true' or 'false', or an Array[Int]"
+  }
+
+  override def validateValue = {
+    case WdlBoolean(value) => ContinueOnReturnCodeFlag(value).successNel
+    case WdlString(value) if value.toLowerCase == "true" => ContinueOnReturnCodeFlag(true).successNel
+    case WdlString(value) if value.toLowerCase == "false" => ContinueOnReturnCodeFlag(false).successNel
+    case WdlInteger(value) => ContinueOnReturnCodeSet(Set(value)).successNel
+    case WdlArray(wdlType, seq) =>
+      val nels: Seq[ErrorOr[Int]] = seq map validateInt
+      val numFailures: Int = nels.count(_.isFailure)
+      if (numFailures == 0) {
+        val defaultReturnCodeNel = Set.empty[Int].successNel[String]
+        nels.foldLeft(defaultReturnCodeNel)((acc, v) => (acc |@| v) { (a, v) => a + v }) map ContinueOnReturnCodeSet
+      } else {
+        failureWithMessage
+      }
```

Comment:
@kshakir FWIW the best textual overview I've seen was the underscore advanced scala book, although it's now cats based. Most of what I know (which is admittedly much less than @mcovarr, but he's the one who has been learning Haskell) comes from watching talks and reading code from the notable scalawags (who also have generally morphed over to cats)